test/Main.purs

```diff
@@ -479,6 +481,29 @@ main = runTest do
       almostEqual' (30.0 .* degree) (asin (scalar 0.5))
       almostEqual' (scalar 1.0) (sin ((0.5 * Number.pi) .* radian))
 
+      for_ [sin /\ csc, cos /\ sec, tan /\ cot, sinh /\ csch, cosh /\ sech, tanh /\ coth] \(f /\ f') -> do
+         let almostEqual'' a b =
+               case a, b of
+                   Left _, _ → failure "Conversion error"
+                   _, Left _ → failure "Conversion error"
+                   Right a', Right b' → almostEqual a' b'
+         let number = scalar 1.3
+
+         almostEqual'' (f number) (map ((scalar 1.0) ⊘ _) (f' number))
+         almostEqual'' (f' number) (map ((scalar 1.0) ⊘ _) (f number))
+
+      for_ [sin /\ asin, cos /\ acos, tan /\ atan] \(f /\ f') -> do
+         let number = scalar 0.4314
+
+         almostEqual' number (f number >>= f')
+         almostEqual' number (f' number >>= f)
+
+      for_ [sec /\ asec, csc /\ acsc, cot /\ acot, sinh /\ asinh, cosh /\ acosh, tanh /\ atanh, sech /\ asech, csch /\ acsch, coth /\ acoth] \(f /\ f') -> do
+         let number = scalar 1.0
+
+         almostEqual' number (f number >>= f')
+         almostEqual' number (f' number >>= f)
```

Comment:
I separated this from the above test since at least some of the functions in this list fail with 0.4314 or its inverse, and I couldn't go to the effort of figuring out which.

I'm not a fan of that, because 1 isn't exactly a number that's hard to trip on even if the function definitions are incorrect. Also, there is no way to know which pairs of functions failed the test (this also applies to the test above).

Reply:
> I separated this from the above test since at least some of the functions in this list fail with 0.4314 or its inverse, and I couldn't go to the effort of figuring out which.

It fails directly for the first example (and possibly others): `sec` and `asec`. The problem is this: 0.4314 is not necessarily in the domain of the function under test.  `asec(x)`, for example, is only defined for

![image](https://user-images.githubusercontent.com/4209276/175812862-0161cd18-46eb-47ba-a7e2-5beb50cf8e81.png)

![image](https://user-images.githubusercontent.com/4209276/175812874-5d712803-5c4b-4e64-8949-a682b4dd8c1d.png)

Also, we cannot simply choose a number larger than 1, because that will lead to problems with other functions, like `atanh`. So we will need to write a bit more custom-tailored tests for these functions, unfortunately.

Reply:
Thank you for the explanation.

I went ahead and figured out exactly which pairs of functions fail the test, and then tested each pair separately.